app/Policies/ServerPolicy.php

```diff
@@ -40,6 +40,16 @@ public function before(User $user, string $ability, string|Server $server): ?boo
             return false;
         }
 
+        // Check if user has role-based permission for this specific ability
+        $permissionName = $ability . ' ' . $this->modelName;
```

Comment:
```suggestion
        $permissionName = "$ability $this->modelName";
```

Isn't there a better way to do this @Boy132 ?